clang/test/Lexer/char8_t.cpp

```diff
@@ -5,6 +5,7 @@
 // RUN: %clang_cc1 -std=c++17 -verify %s
 // RUN: %clang_cc1 -std=c++17 -verify %s -fno-char8_t
 // RUN: %clang_cc1 -std=c++20 -verify %s -fno-char8_t
+// RUN: %clang_cc1 -x c -verify %s -fchar8_t
```

Comment:
Shouldn't this generate a diagnostic now? Or are we planning to continue to accept in cc1 mode? If so, we need a driver test showing we reject from the driver side.

Reply:
`ShouldParseIf` seems never to diagnose. I wanted to ask you about that because it might be surprising
https://godbolt.org/z/6oaPfoWPj

Reply:
CC @jansvoboda11 @MaskRay for more opinions as driver and options maintainers.

I guess I find that behavior kind of surprising. I would expect "you passed this flag and this flag does nothing" should at least be a warning. It's a bit different from an unknown flag, but the same general logic applies: the user passed something and we either know about it and explicitly don't do anything with it, or we don't know about it and don't do anything with it, but either way it seems like the user should be told "this was unknown".

Reply:
I agree, but I think it's preexisting / orthogonal

Reply:
Yup, it is! Only action item I expect out of that is filing an issue if the options folks agree.

Reply:
Filing an issues sounds good to me. I don't think great diagnostics for `-cc1` flags are a high priority, but improving them certainly doesn't hurt!

Reply:
If we want to emit a warning, in RenderCharacterOptions, we cannot unconditionally `claim` the `Arg`, but only claim it for C++.

Reply:
#139058